<launch>
  <arg name="joystick" 		default="true"/>
  <arg name="controller" 	default="true"/>
  <arg name="arduino" 		default="true"/>

  <group if="$(arg joystick)">
    <node name="joystick" pkg="joy" type="joy_node" required="true">
      <param name="dev" value="/dev/input/js2"/>
    </node>
  </group>

  <group if="$(arg controller)">
    <node name="controller3" pkg="joy_control" type="controller3" output="screen" required="true">
      <param name="MAX_Drive_PWM" value="255"/>
      <param name="MAX_Steer_PWM" value="235"/> <!-- 240くらいからオーバーシュートする -->
      <param name="FRIQUENCY" value="100"/>
      <param name="STRKP" value="140"/> <!-- 150以上だと少しオーバーシュートする -->>
      <param name="STRKI" value="0.01"/> 
      <param name="STRKD" value="0.0"/>
      <param name="DRVKP" value="200"/>
      <param name="DRVKI" value="0.25"/>
      <param name="DRVKD" value="0.5"/>
      <param name="STROFFSET" value="10"/> 
      <param name="RADIUS" value="0.133414"/>
      <param name="KV" value="1.0"/>
      <param name="KW" value="5.0"/>
    </node>
  </group>

  <group if="$(arg arduino)">
    <node pkg="rosserial_python" type="serial_node.py" name="arduino" required="true">
      <param name="port" value="/dev/ttyACM0"/>
      <param name="baud" value="57600"/>
    </node>
  </group>
</launch>